geomesa-feature/src/main/scala/org/locationtech/geomesa/feature/serialization/NullableReader.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2015 Commonwealth Computer Research, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.locationtech.geomesa.feature.serialization
+
+import java.util.Date
+
+/** A collection of [[DatumReader]]s for reading objects which may be null.
+  *
+  * Created by mmatz on 4/7/15.
+  */
+trait NullableReader extends PrimitiveReader {
+
+  /**
+   * Reads a null or not-null and if not-null then the value.
+   */
+  def readOption[T](readRaw: DatumReader[T]): DatumReader[Option[T]]
```

Comment:
i would get rid of reading and writing options - it just creates extra objects. do we have a use case where we actually need an option?